### PR TITLE
System Test: Module (WAZUH-DB) -Agent registered with group Y - group data is synced in cluster

### PR DESCRIPTION
**Pull request by Deblintrake09**
In this PR the test suite for `test_assign_agent_group_with_enrollment.py`  is added. A total of 1 new test cases cover that the agent enrollment process and new group assignment works as expected in a cluster environment.

<table>
  <tr>
    <th>Related issues</th>
    <th><a href="[url](https://github.com/wazuh/wazuh-qa/issues/2510)">#2510</a></th>
    <th><a href="[url](https://github.com/wazuh/wazuh/issues/10771)">#10771</a></th>
  </tr>
</table>

## Cases:

- [x] Agent forced to change group:
        - The agent  registered in `master` node with a specific group and it is synced to worker nodes

--- 

## Configuration options
In order to run the test, first the environment located in `/test/system/provisioning/basic_cluster` must be enabled with:
`sudo ansible-playbook -i inventory.yml playbook.yml --extra-vars='{"wazuh_branch": "dev-10771-agent-groups-files-to-wazuh-db"}'`

**Local Internal Options:** No Local Internal Options used

---

## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
| test/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py  | CentOS - Manager | Local | [:large_blue_circle:][:large_blue_circle:][:large_blue_circle:]| 16/03/2022 |  @Deblintrake09   | 

---
- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+
+"""
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py:2]:
```suggestion
"""
```

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+type: system
+brief: Check that when an agent pointing to a worker node is registered using enrrolment method and with 
+       group the change is sync with the cluster.
+tier: 0
+modules:
+    - cluster
+components:
+    - manager
+    - agent
+path: /tests/system/test_cluster/test_agent_groups/test_assign_agent_to_a_group_api.py
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py:15]: Please, check the path.

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+references:
+    - https://github.com/wazuh/wazuh-qa/issues/2510
+tags:
+    - cluster
+"""
+
+import os
+import time
+
+import pytest
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py:48]:
```suggestion
import time
import pytest
```

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                              'provisioning', 'enrollment_cluster', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+tmp_path = os.path.join(local_path, 'tmp')
+id_group = 'group_test'
+enrollment_group = """ 
+    <enrollment>
+    <groups>group_test</groups>
+    </enrollment>
+                       """
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py:69]: `id_group` has the same name as the group in the `enrollment_group` section. Is it possible to use this variable here?

File: tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+@pytest.fixture(scope='function')
+def clean_environment():
+
+    clean_cluster_logs(test_infra_agents + test_infra_managers, host_manager)
+
+    yield
+    # Remove the agent once the test has finished
+    remove_cluster_agents(test_infra_managers[0], test_infra_agents, host_manager)
```
Comment (damarisg) [inline tests/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py:78]: we only required to call it as a parameter. Remove this method defined from here.

**Review by damarisg — CHANGES_REQUESTED** [review]

**Comment by CamiRomero** [comment]
## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
| test/system/test_cluster/test_agent_groups/test_assign_agent_group_with_enrollment.py  | CentOS - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8365072/R1.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8365073/R2.zip)[:green_circle:](https://github.com/wazuh/wazuh-qa/files/8365074/R3.zip)| 28/03/2022 |  @CamiRomero    |

**Review by damarisg — APPROVED** [review]

**Pull request merged**